src/components/general/TextInput/index.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import styles from './styles.less';
+import classNames from 'classnames';
+
+type TextInputProps = {
+    disabled?: boolean;
+    error?: boolean;
+    errorMessage?: string;
+    isRequired?: boolean;
+    placeholder?: string;
+    label?: string;
+    onChange: (newValue: string) => void;
+    value?: string;
+    icon?: any;
+    onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
+    onIconAction?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
+};
+
+const TextInput = React.forwardRef<HTMLDivElement | null, React.PropsWithChildren<TextInputProps>>(
+    (
+        {
+            disabled = false,
+            error,
+            errorMessage,
+            isRequired = false,
+            placeholder = '',
+            label,
+            onChange,
+            value = '',
+            icon,
+            onBlur,
+            onIconAction,
+            ...props
+        },
+        ref
+    ) => {
+        const inputRef = React.useRef<HTMLInputElement>(null);
+        const [focus, setFocus] = React.useState(false);
+        const inputContainerRef = ref as React.RefObject<HTMLDivElement | null>;
+
+        const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+            if (!disabled) {
+                const newValue = event.target.value;
+                onChange(newValue);
+            }
+        };
+
+        const setInputFocus = () => {
+            if (inputRef.current && !disabled) {
+                inputRef.current.focus();
+                setFocus(true);
+            }
+        };
+        const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+            if (!disabled) {
+                setFocus(false);
+                onBlur && onBlur(event);
+            }
+        };
+
+        const inputLabel = label && (
+            <span className={styles.label}>
+                {label}
+                {isRequired && '*'}
+            </span>
+        );
+
+        const inputIcon = icon && (
+            <div className={styles.icon} onClick={onIconAction}>
+                {icon}
+            </div>
+        );
+
+        const inputHelperText = React.useMemo(() => {
+            if (errorMessage && error) {
```

Comment:
This would hide the required help text when an error is present 🤔 Is this by design?

Reply:
This is by design